src/V3Options.cpp

```diff
@@ -917,639 +1108,544 @@ void V3Options::parseOptsList(FileLine* fl, const string& optdir, int argc, char
+    DECL_OPTION("+librescan", CbCall, []() {});  // NOP
+    DECL_OPTION("+notimingchecks", CbCall, []() {});  // NOP
+    DECL_OPTION("+incdir+", CbPartialMatch,
+                [this, &optdir](const char* optp) { addIncDirUser(parseFileArg(optdir, optp)); });
+    DECL_OPTION("+systemverilogext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1800_2017); });
+    DECL_OPTION("+verilog1995ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1364_1995); });
+    DECL_OPTION("+verilog2001ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1364_2001); });
+    DECL_OPTION("+1364-1995ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1364_1995); });
+    DECL_OPTION("+1364-2001ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1364_2001); });
+    DECL_OPTION("+1364-2005ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1364_2005); });
+    DECL_OPTION("+1800-2005ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1800_2005); });
+    DECL_OPTION("+1800-2009ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1800_2009); });
+    DECL_OPTION("+1800-2012ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1800_2012); });
+    DECL_OPTION("+1800-2017ext+", CbPartialMatch,
+                [this](const char* optp) { addLangExt(optp, V3LangCode::L1800_2017); });
+
+    // Minus options
+    DECL_OPTION("-E", Set, &m_preprocOnly);
+    DECL_OPTION("-MMD", OnOff, &m_makeDepend);
+    DECL_OPTION("-MP", OnOff, &m_makePhony);
+    DECL_OPTION("-P", Set, &m_preprocNoLine);
+    DECL_OPTION("-assert", OnOff, &m_assert);
+    DECL_OPTION("-autoflush", OnOff, &m_autoflush);
+    DECL_OPTION("-bbox-sys", OnOff, &m_bboxSys);
+    DECL_OPTION("-bbox-unsup", CbOnOff, [this](bool flag) {
+        m_bboxUnsup = flag;
+        FileLine::globalWarnOff(V3ErrorCode::E_UNSUPPORTED, true);
+    });
+    DECL_OPTION("-build", Set, &m_build);
+    DECL_OPTION("-cc", CbCall, [this]() {
+        m_outFormatOk = true;
+        m_systemC = false;
+    });
+    DECL_OPTION("-cdc", OnOff, &m_cdc);
+    DECL_OPTION("-coverage", CbOnOff, [this](bool flag) { coverage(flag); });
+    DECL_OPTION("-coverage-line", OnOff, &m_coverageLine);
+    DECL_OPTION("-coverage-toggle", OnOff, &m_coverageToggle);
+    DECL_OPTION("-coverage-underscore", OnOff, &m_coverageUnderscore);
+    DECL_OPTION("-coverage-user", OnOff, &m_coverageUser);
+    DECL_OPTION("-debug-abort", CbCall,
+                V3Error::vlAbort);  // Undocumented, see also --debug-sigsegv
+    DECL_OPTION("-debug-check", OnOff, &m_debugCheck);
+    DECL_OPTION("-debug-collision", OnOff, &m_debugCollision);  // Undocumented
```

Comment:
Perhaps we should add an undocumented flag, so we don't use for suggestions?  One way to do this is have DECL_OPTION return a reference that you can call a method on.

```suggestion
    DECL_OPTION("-debug-collision", OnOff, &m_debugCollision).undocumented();
```

Reply:
83d107edebfd2071e929392ca24c1b8e27e608bf adds  undocumented flag.

The actual supression is following line in 2e1da1d8cdd8ba036d1f31af3c6d3d0237762264.
https://github.com/verilator/verilator/blob/2e1da1d8cdd8ba036d1f31af3c6d3d0237762264/src/V3Options.cpp#L216